Tests/SonarScanner.MSBuild.Tasks.UnitTests/GetAnalyzerSettingsTests.cs

```diff
@@ -304,6 +304,152 @@ public void GetAnalyzerSettings_ConfigExists_NewBehaviour_SettingsMerged()
                 "original.should.be.preserved.txt");
         }
 
+        [TestMethod]
+        public void GetAnalyzerSettings_ConfigExists_Legacy_TestProject_SonarAnalyzerSettingsUsed_CSharp()
+        {
+            // Arrange and Act
+            var executedTask = ExecuteGetAnalyzerSettings_ConfigExists_TestProject_SonarAnalyzerSettingsUsed("7.3", "cs");
+
```

Comment:
There's quite a lot of setup required for the new tests so I factored out a helper method with two variables - the language, and the version of SonarQube (we don't expect the behaviour for test projects to change will different versions of SQ, but it does for non-test projects).